src/components/UserProfile/index.jsx

```diff
@@ -33,6 +32,10 @@ class UserProfile extends React.Component {
     this.onVisibilityChange = this.onVisibilityChange.bind(this);
   }
 
+  componentDidMount() {
```

Comment:
So once we load the actual "profile page" part of our app, we then know what the route is and can try to load the user profile for the given user.  

Now that I think about it, we'll need some sort of redirect-based error handling if we're unable to load the profile (bad username, for instance).  This handling should go in the saga, since it catches errors from the backend.